File: python-prov/src/space_in_prefix.py
from prov.model import ProvDocument

from test_case import TestCase


class SpaceInPrefix(TestCase):

    def __init__(self):
        self.filename = "space_in_prefix"

    def create_document(self):
        document = ProvDocument()
        ns = document.add_namespace("ex ex", "https://example.org/")
        ns = document.add_namespace("ex", "https://example.com/")
        document.entity("ex ex:e")
        document.activity("ex:a")
        return document

File: java-prov/src/main/java/cz/muni/fi/bthesis/SpaceInPrefix.java
package cz.muni.fi.bthesis;

import org.openprovenance.prov.model.Activity;
import org.openprovenance.prov.model.Entity;
import org.openprovenance.prov.model.Namespace;
import org.openprovenance.prov.vanilla.Document;
import org.openprovenance.prov.vanilla.ProvFactory;


public class SpaceInPrefix extends TestCase {

    public SpaceInPrefix() {
        setFilename("space_in_prefix");
    }

    @Override
    public Document createDocument() {
        var factory = new ProvFactory();
        var document = new Document();
        var ns = new Namespace();
        ns.addKnownNamespaces();
        ns.register("ex ex", "https://example.org/");
        ns.register("ex", "https://example.com/");
        var e = ns.qualifiedName("ex ex", "e", factory);
        var a = ns.qualifiedName("ex", "a", factory);
        Entity entity = factory.newEntity(e);
        Activity activity = factory.newActivity(a);
        document.getStatementOrBundle().add(entity);
        document.getStatementOrBundle().add(activity);
        document.setNamespace(ns);
        return document;
    }
}
